Revise frozen Safari UA iOS version to match actual last-shipped iOS version

brentfulgham opened the pull request:
#### 030bd59b89c422ddc09e632013d6d7b579c43254
<pre>
Revise frozen Safari UA iOS version to match actual last-shipped iOS version
<a href="https://bugs.webkit.org/show_bug.cgi?id=298473">https://bugs.webkit.org/show_bug.cgi?id=298473</a>
&lt;<a href="https://rdar.apple.com/159955141">rdar://159955141</a>&gt;

Reviewed by Matthieu Dubet.

Safari froze the iOS version number in it&apos;s UA string to address compatibility issues
encountered when we attempted to report iOS 26 versioning. That change was done in
an internal repository.

This change moves the iOS version freezing logic to Open Source, and corrects the
version to match that last iOS 18.6.2 patch level.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::standardUserAgentWithApplicationNameIncludingCompatOverrides):

Canonical link: <a href="https://commits.webkit.org/299765@main">https://commits.webkit.org/299765@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6ad1a2382e5ae09692fcc522f47323dc2e3f64ff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120106 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30451 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126451 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72179 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ad318b99-8ace-4765-bcdb-8fca43b1f5d6) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121982 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40496 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48377 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91215 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/60522 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/23328132-75de-4454-b36d-94d9a24fd090) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123058 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32347 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71766 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/5fd845f0-7a73-4726-a8d9-7e56b7a4029a) 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25796 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70078 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101826 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25983 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129359 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47026 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35672 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99833 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47392 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103878 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99676 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25308 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45127 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43657 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46888 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52594 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46356 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49703 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48040 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

burg commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1570,11 +1570,32 @@ bool Quirks::needsLaxSameSiteCookieQuirk(const URL& requestURL) const
 String Quirks::standardUserAgentWithApplicationNameIncludingCompatOverrides(const String& applicationName, const String& userAgentOSVersion, UserAgentType type)
 {
-    return standardUserAgentWithApplicationNameIncludingCompatOverridesInternal(applicationName, userAgentOSVersion, type);
+    auto overridenUAString = standardUserAgentWithApplicationNameIncludingCompatOverridesInternal(applicationName, userAgentOSVersion, type);
```
Nit: overridden

brentfulgham replied:
Ugh! Ugh ugh ugh!

brentfulgham replied:
Fixing!

mdubet commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1570,11 +1570,32 @@ bool Quirks::needsLaxSameSiteCookieQuirk(const URL& requestURL) const
+        return overriddenUAString;
+
+    if (userAgentOSVersion == frozenVersion)
+        return String();
```
`return { }` would be more generic

brentfulgham replied:
Will fix.

mdubet commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -1570,11 +1570,32 @@ bool Quirks::needsLaxSameSiteCookieQuirk(const URL& requestURL) const
+#elif PLATFORM(WATCHOS)
+static constexpr auto frozenVersion = "11_6_1"_s;
+#elif PLATFORM(APPLETV)
+static constexpr auto frozenVersion = "18_6_2"_s;
```
Do we need the special case for APPLETV (it seems to be the same UA than the fallback)

brentfulgham replied:
We don't today, but we might in future. Might as well leave them distinct for now.

brentfulgham replied:
Actually, we do! TV OS is still on 18.6 (not 18.6.2). I've corrected this.

brentfulgham committed:
Revise frozen Safari UA iOS version to match actual last-shipped iOS version
https://bugs.webkit.org/show_bug.cgi?id=298473
<rdar://159955141>

Reviewed by Matthieu Dubet.

Safari froze the iOS version number in it's UA string to address compatibility issues
encountered when we attempted to report iOS 26 versioning. That change was done in
an internal repository.

This change moves the iOS version freezing logic to Open Source, and corrects the
version to match that last iOS 18.6.2 patch level.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::standardUserAgentWithApplicationNameIncludingCompatOverrides):

Canonical link: https://commits.webkit.org/299765@main
